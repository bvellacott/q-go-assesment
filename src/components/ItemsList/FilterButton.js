const FilterButton = () => { throw new Error('not implemented') }
