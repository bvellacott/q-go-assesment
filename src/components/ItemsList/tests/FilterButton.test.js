import React from 'react'
import { mount } from 'enzyme'
import FilterButton from '../FilterButton'

describe('filter button', () => {
	it('Sets button text', () => {
		const wrapper = mount(<FilterButton>test filter</FilterButton>)
		const filterButton = wrapper.find('.itemList_filterButton')
		expect(filterButton.length).toEqual(1)
		expect(filterButton.text()).toEqual('test filter')
	})

	it('calls setFilter on click', () => {
		const setFilter = jest.fn()
		const wrapper = mount(<FilterButton setFilter={setFilter} />)
		const filterButton = wrapper.find('.itemList_filterButton')
		expect(filterButton.length).toEqual(1)
		filterButton.simulate('click')
		expect(setFilter).toHaveBeenCalled()
	})
})