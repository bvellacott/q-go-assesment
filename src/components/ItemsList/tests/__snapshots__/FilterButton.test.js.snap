// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter button Sets button text 1`] = `
<FilterButton>
  <button
    className="itemList_filterButton"
    onClick={undefined}
  >
    test filter
  </button>
</FilterButton>
`;

exports[`filter button calls setFilter on click 1`] = `
<FilterButton
  setFilter={[Function]}
>
  <button
    className="itemList_filterButton"
    onClick={[Function]}
  />
</FilterButton>
`;
